src/state_machine/mod.rs

```diff
@@ -13,6 +13,7 @@ use std::fmt::Debug;
 mod channel;
 mod null;
 mod register;
+#[cfg(feature="hashmap")] mod hashmap;
```

Comment:
this either needs to be pub mod, or reexported as the ones below are for anyone to use it.